api/envoy/extensions/original_ip_detection/xff/v3/xff.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.xff.v3;
+
+import "envoy/extensions/original_ip_detection/common/v3/original_ip_detection.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.xff.v3";
+option java_outer_classname = "XffProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: XFF original IP detection extension]
+
+// This extension allows for the original downstream remote IP to be detected
+// by reading the :ref:`config_http_conn_man_headers_x-forwarded-for` header.
+//
+// If no other extension is configured or detection fails for the configured
+// extension, Envoy will fallback to using this extension.
```

Comment:
Per above I think this is odd. Explicit repeated list?

Reply:
Oh, I think I see what you mean now -- it'd look and behave like this:

```
repeated config.core.v3.TypedExtensionConfig original_ip_detection = ...
```

which by default, gets init to a list of just `[XFF extension]`. But if the user overrides this list, XFF is still appended to that list.

I am OK with this, we just have to extend the logic to handle a list of extensions and the possibility of rejecting the request early.

@alyssawilk @antoniovicente: would that work for your use cases?

Reply:
> But if the user overrides this list, XFF is still appended to that list.

This is what I'm suggesting we don't do. If someone wants XFF as a fallback they can explicitly configure it? I'm not sure why we would do magic fallback when this is all opt in?

Reply:
Ah, gotcha -- yes... we are not really preserving any behavior there.